selftest/cgroup/main.go

```diff
@@ -0,0 +1,81 @@
+package main
+
+import "C"
+
+import (
+	"encoding/binary"
+	"os"
+	"os/exec"
+
+	"fmt"
+
+	bpf "github.com/aquasecurity/libbpfgo"
+)
+
+func main() {
+	bpfModule, err := bpf.NewModuleFromFile("main.bpf.o")
+	if err != nil {
+		fmt.Fprintln(os.Stderr, err)
+		os.Exit(-1)
+	}
+	defer bpfModule.Close()
+
+	err = bpfModule.BPFLoadObject()
+	if err != nil {
+		fmt.Fprintln(os.Stderr, err)
+		os.Exit(-1)
+	}
+
+	cgroupRootDir := "/sys/fs/cgroup/unified"
+	if _, err := os.Stat(cgroupRootDir); os.IsNotExist(err) {
+		cgroupRootDir = "/sys/fs/cgroup"
+	}
+
+	prog, err := bpfModule.GetProgram("cgroup__socket")
+	if err != nil {
+		fmt.Fprintln(os.Stderr, err)
+		os.Exit(-1)
+	}
+
+	link, err := prog.AttachCgroup(cgroupRootDir)
+	if err != nil {
+		fmt.Fprintln(os.Stderr, err)
+		os.Exit(-1)
+	}
+	if link.GetFd() == 0 {
+		os.Exit(-1)
+	}
+
+	eventsChannel := make(chan []byte)
+	rb, err := bpfModule.InitRingBuf("events", eventsChannel)
+	if err != nil {
+		fmt.Fprintln(os.Stderr, err)
+		os.Exit(-1)
+	}
+
+	rb.Start()
+	numberOfEventsReceived := 0
+	go func() {
+		_, err := exec.Command("ping", "localhost", "-c 10").Output()
```

Comment:
This will create a single socket and test needs 5 socket creations, right ? We should spawn `ping localhost -c 1 -w 1` for 5 times instead ? Or something like it...

Reply:
fixed